spec/controllers/users_controller_spec.rb

```diff
@@ -39,13 +39,61 @@
     it { expect(response).to redirect_to(users_path) }
   end
 
+  describe '#edit' do
+    let(:name) { 'Test Users' }
+    let(:email) { 'test@example.com' }
+
+    let(:provider) { Kuroko2::User::GROUP_PROVIDER }
+    let(:user) do
+      create(
+        :user,
+        name: 'name',
+        email: 'initial@example.com',
+        provider: provider,
+      )
+    end
+
+    before { put(:update,  params: { id: user.id, user: { name: name, email: email } }) }
+
+    it 'edits the name/email' do
+      expect(assigns(:user).reload.name).to eq(name)
+      expect(assigns(:user).reload.email).to eq(email)
+      expect(response).to redirect_to(user_path(user))
+    end
+
+    context 'if user is google account' do
+      let(:provider) { Kuroko2::User::GOOGLE_OAUTH2_PROVIDER }
+
+      it 'returns bad request' do
+        expect(response).to be_bad_request
```

Comment:
[IMO] I prefer to use `to have_http_status(:bad_request)` because it provides more readable failure message